lib/imports/api/authority-registry/server/authority-registry.js

```diff
@@ -0,0 +1,319 @@
+      insertAuthority.call({ ip: this.ip, port: this.port }, (err, res) => {
+        if (err) {
+          return cb(new Error(`Failed to insert Authority server into registry: ${err}`));
+        }
+        console.log(`ProseMeteor AuthorityRegistry: Registered server in authority registry: ${this.ip}:${this.port}`);
+        return cb(null, res);
+      });
+      // at the same time, start the health checks on all other servers
+      this.runHealthChecks(this.fullRegistry());
+    };
+    // check if this sever's already registered first
+    if (this.isRegistered({ ip: this.ip, port: this.port })) {
+      console.log('ProseMeteor AuthorityRegistry: Authority server was already registered, clearing it out');
+      removeAuthority.call({ ip: this.ip }, (err, res) => {
+        if (err) {
+          return cb(new Error(`Failed to remove existing authority server ${err}`));
+        }
+        return insert();
+      });
+    } else {
+      return insert();
+    }
+  }
+  /**
+  *  Remove an app server from the registry, because it's no longer available to host document authorities.
+  * @param {Object} params
+  * @param {String} params.ip     ip of the server to remove from the registry
+  * @param {Number} params.port   port of the server to remove from the registry
+  * @param {Function} cb          callback
+  */
+  unregister ({ ip, port }, cb) {
+    check(ip, String);
+    check(port, Number);
+    check(cb, Function);
+    // remove from collection
+    removeAuthority.call({ ip, port }, (err, res) => {
+      if (err) {
+        return cb(new Error(`Failed to remove Authority server from registry: ${err}`));
+      }
+      console.log(`ProseMeteor AuthorityRegistry: Unregistered authority server with ip ${ip}`);
+      return cb(null, res);
+    });
+  }
+
+  /**
+  * Runs a health check against a single server.
+  * @param {Object} params
+  * @param {String} params.authorityUrl     url of the authority
+  * @param {Function} cb                  callback, given Boolean of if health check passed
+  */
+  runSingleHealthCheck ({ authorityUrl }, cb) {
+    const authority = this.getAuthorityInfoByUrl({ authorityUrl });
+    // run health check. if number of failed === 0, it was healthy
+    const numFailed = this.runHealthChecks([ authority ]);
+    const passed = numFailed === 0;
+    console.log(`ProseMeteor AuthorityRegistry: ran single health check for ${authorityUrl}, passed=${passed}, numFailed=${numFailed}`);
+    return passed;
+  }
+  /**
+  *  Run health checks against every server in the registry, keeping track of which have failed
+  * and removing those that have failed more than the threshold.
+  * @param {Array} serversArray   array of server objects with { _id, {String} ip, {Number} port, {Number} failedHealthChecks}
+  * @param {Function} cb          callback, with number of failed connections passed to it
+  * @returns {Number} number of failed connections
+  */
+  runHealthChecks (serversArray) {
```

Comment:
moved health check logic to its own module that exports functions for use in `ProseMeteorAuthorityRegistry`, commits incoming